File: abstractions/typescript/src/authentication/authenticationProvider.ts
import { RequestInformation } from "../requestInformation";

/**
 * @interface
 * Interface to be implementated to provide authentication information for a request.
 * @property {Function} authenticateRequest - The function to authenticate the request.
 */
export interface AuthenticationProvider {
  /**
   * Authenticates the application and returns a token base on the provided Uri.
   * @param request the request to authenticate.
   * @return a Promise to await for the authentication to be completed.
   */
  authenticateRequest: (request: RequestInformation) => Promise<void>;
}

File: abstractions/python/kiota/abstractions/authentication/authentication_provider.py
from abc import ABC, abstractmethod

from ..request_information import RequestInformation


class AuthenticationProvider(ABC):
    """
    Base class for providing authentication information for a request.
    """

    @abstractmethod
    async def authenticate_request(self, request: RequestInformation) -> None:
        """Authenticates the application request

        Args:
            request (RequestInformation): The request to authenticate
        """
        pass
